server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -469,7 +469,8 @@ private boolean isSystemIndex(IndexMetadata indexMetadata) {
                     .filter(requestedDataStreams::contains)
                     .collect(Collectors.toList());
                 if (intersectingDataStreams.isEmpty() == false) {
-                    dataStreamAliases.put(alias.getName(), new DataStreamAlias(alias.getName(), intersectingDataStreams));
+                    DataStreamAlias copy = new DataStreamAlias(alias.getName(), intersectingDataStreams, alias.getWriteDataStream());
+                    dataStreamAliases.put(alias.getName(), copy);
```

Comment:
I'm not especially familiar with the snapshot/restore code, but does this work if some of the alias's data streams are present in `intersectingDataStreams` but the alias's write data stream is not? Also, it looks like it would drop data streams from the alias if they were already present in the cluster but not part of the restore?

Reply:
Yes, that is right. I need to fix that (and add a test for it). I wasn't completely finished with this pr yet (meant to open this as a draft pr, but forgot to select it as such).

Reply:
> Also, it looks like it would drop data streams from the alias if they were already present in the cluster but not part of the restore?

This code just selects the data streams with aliases from the snapshot to restore. As part of the `RestoreSnapshotStateTask` this is merged with the cluster state the snapshot is restored into (see `applyDataStreamRestores()` method).